client/src/partials/services/services.js

```diff
@@ -1,183 +1,145 @@
+    refreshServices();
 
-  function setAction (value, service) {
-    $scope.choosen = angular.copy(service) || {};
-    if (value === 'more') {
-      getCost($scope.choosen.cost_center_id)
-      .then(handleResultCost)
-      .then(getProfit)
-      .then(handleResultProfit);
-    } else if (value === 'edit') {
-      configuration.cost_centers = $scope.model.costs.data;
-      configuration.profit_centers = $scope.model.profits.data;
-    }
-    $scope.action = value;
-  }
+    // load Enterprises
+    Enterprises.read().then(function (data) {
+      vm.enterprises = data;
+    }).catch(handler);
 
-  function getProfit() {
-    return connect.req('/profit/' + $scope.project.id + '/' + $scope.choosen.profit_center_id);
-  }
+    // load Cost Center
+    FinancialService.readCostCenter().then(function (data) {
+      vm.costCenters = data;
+    }).catch(handler);
 
-  function edit() {
-    var data = {
-      id                : $scope.choosen.id,
-      name              : $scope.choosen.service,
-      cost_center_id    : $scope.choosen.cost_center_id,
-      profit_center_id  : $scope.choosen.profit_center_id
-    };
-
-    if (isValid(data.cost_center_id, data.profit_center_id)) {
-      connect.put('service', [connect.clean(data)], ['id'])
-      .then(function () {
-        $scope.choosen.cost_center = $scope.choosen.cost_center_id ? getCostcenterText($scope.choosen.cost_center_id) : $scope.choosen.cost_center;
-        $scope.choosen.profit_center = $scope.choosen.profit_center_id ? getProfitCenterText($scope.choosen.profit_center_id) : $scope.choosen.profit_center;
-        $scope.model.services.put(connect.clean($scope.choosen));
-        $scope.action = '';
-        $scope.choosen = {}; // reset
-      })
-      .catch(function (err) {
-        messenger.danger('Error:' + JSON.stringify(err));
-      });
-    }else{
-      alert('centre de cout ou centre de profit deja utilise');
-    }
+    // load Profit Center
+    FinancialService.readProfitCenter().then(function (data) {
+      vm.profitCenters = data;
+    }).catch(handler);
 
+    vm.state.reset();
   }
 
-  function isValid (cost_id, profit_id) {
-    var c = $scope.model.services.data.filter(function (item) {
-      return (item.cost_center_id === cost_id && $scope.choosen.id !== item.id) || (item.profit_center_id === profit_id && $scope.choosen.id !== item.id);
-    })[0];
-
-    return c ? false : true;
+  function cancel() {
+    vm.state.reset();
+    vm.view = 'default';
   }
 
-  function getCostcenterText(cost_id) {
-    //FIX ME : hack
-    return configuration.cost_centers.filter(function (item) {
-      return item.id === cost_id;
-    })[0].text;
+  function create() {
+    vm.view = 'create';
+    vm.service = {};
   }
 
-  function getProfitCenterText (profit_id) {
-    //FIX ME : hack
-    return configuration.profit_centers.filter(function (item) {
-      return item.id === profit_id;
-    })[0].text;
+  // switch to update mode
+  function update(data) {
```

Comment:
What type of parameter is `data` in the update function?  Is it a number, object, ... ?  Could you leave a comment telling future developers what it is?